src/drt/src/TritonRoute.cpp

```diff
@@ -911,6 +911,19 @@ void TritonRoute::addUserSelectedVia(const std::string& viaName)
   }
 }
 
+void TritonRoute::setUnidirectionalLayer(const std::string& layerName)
+{
+  if (db_->getTech() == nullptr) {
+    logger_->error(DRT, 615, "Load tech before setting unidirectional layers");
+  }
+  auto tech = db_->getTech();
+  if (tech->findLayer(layerName.c_str()) == nullptr) {
+    logger_->error(utl::DRT, 616, "Layer {} not found", layerName);
+  } else {
+    design_->getTech()->setUnidirectionalLayer(layerName);
```

Comment:
Keep the return value of findLayer and use the pointer rather than the name from this point forward.

Reply:
Never mind I see this is before the frLayer is generated

Reply:
still it makes more sense to keep the pointer instead, right?